galasa-parent/dev.galasa.framework.api.authentication/src/main/java/dev/galasa/framework/api/authentication/internal/OidcProvider.java

```diff
@@ -201,15 +201,15 @@ public HttpResponse<String> sendTokenPost(String clientId, String clientSecret,
         // Convert the request's JSON object to the application/x-www-form-urlencoded content type
         // as required by the /token endpoint.
         sbRequestBody.append("grant_type=authorization_code");
-        sbRequestBody.append("&code=" + authCode);
-        sbRequestBody.append("&client_id=" + clientId);
-        sbRequestBody.append("&client_secret=" + clientSecret);
-        sbRequestBody.append("&redirect_uri=" + redirectUri);
+        sbRequestBody.append("&code=" + URLEncoder.encode(authCode, StandardCharsets.UTF_8));
+        sbRequestBody.append("&client_id=" + URLEncoder.encode(clientId, StandardCharsets.UTF_8));
```

Comment:
Is this necessary ? Can't we just validate + fail if it's in the wrong format ? Why jump to mashing the contents with an encoder ?

Reply:
Thanks, we call this method after already having validated (otherwise, we would have already failed), so will remove.